src/main/java/trackr/logic/commands/EditTaskCommand.java

```diff
@@ -43,18 +43,17 @@ public class EditTaskCommand extends Command {
     public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the task list.";
 
     private final Index index;
-    private final EditTaskDescriptor editTaskDescriptor;
+    private final TaskDescriptor taskDescriptor;
 
     /**
      * @param index of the task in the filtered task list to edit
-     * @param editTaskDescriptor details to edit the task with
+     * @param taskDescriptor details to edit the task with
      */
-    public EditTaskCommand(Index index, EditTaskDescriptor editTaskDescriptor) {
-        requireNonNull(index);
-        requireNonNull(editTaskDescriptor);
+    public EditTaskCommand(Index index, TaskDescriptor taskDescriptor) {
+        requireAllNonNull(index, taskDescriptor);
 
         this.index = index;
-        this.editTaskDescriptor = new EditTaskDescriptor(editTaskDescriptor);
+        this.taskDescriptor = new TaskDescriptor(taskDescriptor);
```

Comment:
I think the name of the variable can stay as editTaskDescriptor since this is related to editing tasks?
Otherwise, comment at line 82 should be edited to reflect taskDescriptor